presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -105,6 +108,9 @@ private PlanNode getCostBasedJoin(JoinNode joinNode, Context context)
             if (isSizeBasedJoinDistributionTypeEnabled(context.getSession())) {
```

Comment:
How is this different from sizeBasedJoin here?

Reply:
The size based join only considers the size of input table, even when the estimation of immediate probe/build input is available.

For example, if I have a query like 
with B as (select * from t1 join t2 using (key)) select * from A Join B using (key),
if we have estimation of A unknown, but estimation of B is small, we will not have broadcast with size based join. As in size based join, the size of t1 and t2 considered not representative of size of B after join operation. This is one of the case this PR is trying to solve.

I also thought about patching to the size based join, however, the size based join produces query plans for cases which I do not need here, and I will still need to have a separate session parameter inside the size based join implementation to control it. And it also makes the logic of size based join more complex. Hence I chose to have it as a separate part here.